libs/facilities/frontend/view/src/lib/components/detail/detail.page.ts

```diff
@@ -162,21 +167,69 @@ export class XdDetailPage implements OnInit {
 			},
 		],
 	};
-	protected readonly envChart: Signal<EChartsOption | undefined> = computed(() => {
-		const envData = this.envData();
-		if (!envData) return undefined;
 
-		const envChart = {
-			...this.envOptions,
-		};
+    protected readonly envChart: Signal<EChartsOption | undefined> = computed(() => {
+        const envData = this.envData();
+        if (!envData) return undefined;
 
-		if (!envChart.series || !(envChart.series instanceof Array)) return undefined;
+        const envChart = {
+            ...this.envOptions,
+        };
+
+        if (!envChart.series || !(envChart.series instanceof Array)) return undefined;
+
+        envChart.series[0].data = envData.map((item) => [ item.time, item['Temperature'] ]);
+        envChart.series[1].data = envData.map((item) => [ item.time, item['Humidity'] ]);
+        envChart.series[2].data = envData.map((item) => [ item.time, item['Pressure'] ]);
+        return envChart;
+    });
+
+    protected readonly metricsChart: Signal<EChartsOption | undefined> = computed(() => {
+        const facility = this.facility();
+        if (!facility) return undefined;
+
+        const metrics = facility.metrics;
+
+        const xAxisData = map(metrics, item => PUMP_METRICS_FULL_NAME_MAP[item.name].replace(/ /g, '\n').trim());
+        const seriesKeys = $enum(EMetricsCategory).getValues();
+
+        const seriesData = map(seriesKeys, (key) => ({
+            type: 'bar',
+            name: METRIC_CATEGORY_COLOR_INFORMATION[key].abbreviation,
+            data: map(metrics, item => item[key]),
+            emphasis: { focus: 'series' },
+            itemStyle: { color: METRIC_CATEGORY_COLOR_INFORMATION[key].color },
+        }));
+
+        return {
+            tooltip: {
+                trigger: 'axis'
+            },
+            legend: {
+                top: 30,
+                left: 80,
+            },
+            grid: {
+                top: 80,
+            },
+            title: {
+                text: 'Pump Metrics',
+                left: 'center',
+            },
+            xAxis: {
+                type: 'category',
+                data: xAxisData,
+                nameLocation: 'middle',
+            },
+            yAxis: {
+                type: 'value',
+                nameLocation: 'middle',
+            },
+            series: seriesData,
+            emphasis: { focus: 'series', scaleSize: 80, scale: true },
+        };
+    });
```

Comment:
In the other chart signals I did not return the whole thing { ... } here but I already hat them set and just changed the non static values. I did that because in this PR in the 3rd resolved conversation you told me all that static data should not be inside the compute method https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/pull/181

So idk what is better but it would be nice if it is the same everywhere

Reply:
Yeah your write but this file is already flooded by unlimited data. But yeah ur right I alinged it with your code.

Reply:
yeah this file is very cramped but still thank you for adding it